stack/base/load-singlesshagent.sh

```diff
@@ -0,0 +1,67 @@
+# Make sure you source this script rather than executing it
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]
+then
+    echo "You need to source this script, stopping." >&2
+    exit 1
+fi
+
+# run as
+# source load-singlesshagent.sh -v
+# for verbose output
+
+load_singlesshagent() {
+    local VERBOSE
+    local SSH_ENV
+    local SSH_ADD_OUTPUT
+    local SSHADD_RETVAL
+    local NUMKEYS
+
+    VERBOSE=false
+    if [ "$1" == "-v" ]
+    then
+        VERBOSE=true
+    fi
+
+    [ "$VERBOSE" == "true" ] && echo "Single SSH agent script [verbose mode]" >&2
+    SSH_ENV="$HOME/.ssh/agent-environment"
+    # Source SSH settings, if applicable
+    if [ -r "${SSH_ENV}" ]; then
+        source "${SSH_ENV}" > /dev/null
+        [ "$VERBOSE" == "true" ] && echo "- sourcing existing environment" >&2
+    else
+        [ "$VERBOSE" == "true" ] && echo "- no existing environment to source" >&2
+    fi
+
+    SSH_ADD_OUTPUT=`ssh-add -l 2> /dev/null`
+    # Needed, the later 'test' calls will replace this
+    SSHADD_RETVAL="$?"
+    # Error code: 0: there are keys; 1: there are no keys; 2: cannot contact agent
+    if [ "$SSHADD_RETVAL" == "2" ]
+    then
+        [ "$VERBOSE" == "true" ] && echo "  - unable to contact agent, creating a new one" >&2
+        (umask 066; ssh-agent > ${SSH_ENV})
+        source "${SSH_ENV}" > /dev/null
+    elif [ "$SSHADD_RETVAL" == "1" ]
+    then
+        [ "$VERBOSE" == "true" ] && echo "  - ssh-agent found (${SSH_AGENT_PID}), no keys (I might want to add keys here)" >&2
+        # Maybe the user wants to do it by himself?
+        #ssh-add
```

Comment:
```suggestion
        ssh-add ~/.ssh/id_rsa
```
I'd argue to run this, we create `id_rsa`/`id_rsa.pub` pair in the container start script without passphrase, so we know it can be added without any problem. 
It also makes it possible to use ssh-agent for remote config that doesn't specify `IdentifyFile`.

Reply:
Yes, that sounds good. Please go ahead and change it and add a comment with the explanation.